Sources/Container.swift

```diff
@@ -176,6 +176,23 @@ public final class Container {
     public func addBehavior(_ behavior: Behavior) {
         behaviors.append(behavior)
     }
+    
+    /// Check if a ``Service`` of a given type and name has already been registered.
+    ///
+    /// - Parameters:
+    ///   - serviceType: The service type to compare.
+    ///   - name:        A registration name, which is used to differentiate from other registrations
+    ///                  that have the same service and factory types.
+    ///
+    /// - Returns: A ``Bool`` which represents whether or not the ``Service`` has been registered.
+    public func hasRegistration<Service>(
+        _ serviceType: Service.Type,
+        name: String? = nil
+    ) -> Bool {
+        getRegistrations().contains { key, _ in
+            key.serviceType == serviceType && key.name == name
+        }
```

Comment:
Just thinking about this possible scenario:
```
// What kind of registration?
//
// Could it be helpful to know if it has a 0-arg registration 
// vs an n-arg registration & which types?
if container.hasRegistration(TypeWithArgs.self) {
  container.register( … )
}
```
Maybe you could use the ServiceKey struct here and compare it against `services[key]` instead? Should be a bit more performant too for larger projects.

Reply:
Hi, exactly because of the reason you mentioned here: 
> If you register an identical ServiceKey it will override the existing registration (if there is one)

This is crucial (at least for me). We have lazily loaded assemblies that lives on other modules, and are loaded some time after the main container of the app has registered everything. On this scenario I don't want them to override the current registrations.

I've checked the `ServiceKey` way before implementing it the way I did, but assembling one from scratch just to compare is a tough one.

Reply:
Regarding 0-arg vs n-arg, I don't think it is relevant for the scenario, because those variations are relevant to the implementations of the service and not for it's protocol.

Reply:
Thanks for the quick reply. In that case, perhaps naming the function `hasAnyRegistration` will be more clear.

For your specific scenario I think this is sufficient. But, for the purpose of lazy Assemblies that shouldn't overwrite a _specific_ registration—the natural extension of this PR would be to add `hasRegistration` which could check against a ServiceKey. It could also be quite useful, if you're interested. :)

Reply:
@maxim-chipeev  `ServiceKey` is an internal type, and not exposed to `Swinject`'s consumers (:

Reply:
In any case, I've updated the API as you suggested to: `hasAnyRegistrations(of serviceType)`

Reply:
In any case, I've updated the API as you suggested to: hasAnyRegistrations(of serviceType)

Reply:
right! 😅 I should've put it a different way—to check against the Service type, name, & argument types, this doesn't require exposing the ServiceKey

Reply:
Yeah, so I'm already checking against type and name, but argument types is pretty unintuitive to assemble, so we might need a different set of generic variations, per each argument count, similar to autoRegister.